src/simmate/apps/evolution/stop_conditions/stop_conditions.py

```diff
@@ -21,35 +27,114 @@ def check(self, search):
+        num_new_structures_since_best = search.individuals.filter(
+            energy_per_atom__gt=best.energy_per_atom + search.convergence_cutoff,
+            finished_at__gte=best.finished_at,
+        ).count()
+        if num_new_structures_since_best > search.best_survival_cutoff:
+            logging.info(
+                "Best individual has not changed after "
+                f"{self.best_survival_cutoff} new individuals added."
+            )
             return True
+        # If we reached this point, then we haven't hit a stop condition yet!
+        return False
 
-        # look how long the minimum structure has been there #!!! ERROR IN LOGIC
-        # min_index = energies.index(min_energy)
-        # if len(energies) - min_index >= self.same_min_structures:
-        #     print('\nMin-structure reign limit hit. Stopping Search')
-        #     return True
 
-        # If no conditions above are met, return False to continue the search
-        return False
+class ExpectedStructure(StopCondition):
+    """
+    Stops the search if the provided expected structure has been found.
+    """
+
+    def check(self, search):
+        # !!! Here we compare individuals using pymatgen's StructureMatcher. An
+        # alternative would be to use the searches validator property like we
+        # do to find unique individuals in the search itself. Depending on how
+        # slow the following for loop is, that may be necessary
+        # get expected structure and make sure it is a pymatgen Structure object
+        expected_structure_dynamic = search.expected_structure
+        # make sure we have an expected structure. If not we won't use this stop condition
+        if not expected_structure_dynamic:
+            logging.warning(
+                "No expected_structure was provided. The ExpectedStructure stop condition will be ignored"
+            )
+            return False
+        expected_structure = Structure.from_dynamic(expected_structure_dynamic)
+        # get completed individuals
+        individuals = search.individuals_completed.order_by("finished_at").all()
+        # make sure we have some individuals to compare to. If not, we haven't
+        # finished any structures yet and immedieately return False
+        if len(individuals) == 0:
+            return False
+
+        structures = individuals.to_toolkit()
+
+        # create instance of structure matcher
+        matcher = StructureMatcher(attempt_supercell=True)
+
+        # for each structure, check if it matches the expected structure
+        for n, structure in track(list(enumerate(structures))):
+
+            is_match = matcher.fit(expected_structure, structure)
+            if is_match:
+                # structure.to("cif", d / "match.cif")
+                # We've found our structure and can immediately move on
+                break
+
+        if not is_match:
+            logging.info("Search has not found groundstate yet")
```

Comment:
update the logging to be "expected structure", not "groundstate". My original print statements were groundstate but your ExpectedStructure class is more flexible